### ISSUE TITLE
Reproduction question: hyperparameters for Figure 3 (FCA-C vs VFC)

### ISSUE THREAD

**Issue by cgewalt** [kwkimonline/FCA#1]
Hi, 

we are a group of university students working on a practical course project and are trying to reproduce Figure 3 from your paper "Fair Clustering via Alignment".

We ran our own experiments on the Adult dataset using the following setup:

for eps in $(seq 0.05 0.1 0.9)
do
  python -m src.main \
    --data_name Adult \
    --K 10 \
    --epsilon $eps \
    --l2_normalize \
    --seed 2025
done

However, our results differ significantly from the published Figure 3. 
Since our results differ substantially from the published Figure 3, we wondered whether the original plot might have been generated using different parameter configurations, which could explain the discrepancy. 
We would greatly appreciate it if you could share the exact hyperparameter settings and any additional experimental details (e.g., normalization choices, fixed seeds, or other relevant settings) that were used to produce this figure.
This would help us greatly in accurately reproducing and understanding your results.

Thank you very much for your time!

**Comment by kwkimonline** [comment]
Hi,
Thank you for your interest in this project.

In short, the code (the previous version) was only slightly behind the version used for the camera-ready paper. This was a minor update (it does not change the algorithm itself), and I have now pushed the updated code. The only (main) change that matters for reproducing is that, when updating W in FCA-C, we use all samples (which is also consistent with the paper).

With the updated code, I re-ran the experiments on the Adult dataset with multiple seeds and observed very similar trends with (or sometimes better than) Figure 3. Here's one example run:

| epsilon | balance | cost |
|---|---|---|
0.10 | 0.283 | 0.2980
0.07 | 0.373 | 0.3039
0.05 | 0.408 | 0.3098
0.03 | 0.431 | 0.3147
0.02 | 0.463 | 0.3158
0.01 | 0.489 | 0.3164
0.00 | 0.493 | 0.3161

One small note: because of this change, we may need to try slightly smaller epsilon values than before. Also, the clustering cost can vary a bit depending on the random seed (and randomness in sklearn’s KMeans initialization). However, this does not affect the main message of Figure 3: we can still observe a trade-off similar to VFC, while achieving higher fairness levels.

Please let me know if anything still looks different. I'll close this issue for now, but feel free to reopen it if needed.

Best,
